pkg/portal/cluster/statistics.go

```diff
@@ -0,0 +1,115 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	prometheusAPI "github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+type MetricValue struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     float64   `json:"value"`
+}
+
+type Metrics struct {
+	Name  string        `json:"metricname"`
+	Value []MetricValue `json:"metricvalue"`
+}
+
+func (c *client) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	return c.fetcher.Statistics(ctx, httpClient, promQuery, duration, endTime)
+}
+
+func (f *realFetcher) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
```

Comment:
why is this a method on the real fetcher? you don't use any of its fields

Reply:
hmmm, Actually I wanted to make this `client's` method and since the `client` has `realfetcher` in it so in order to call it from the portal package I did this, but yeah .. I learned from the azure-side-information PR and will try correcting this and let you know.

Reply:
This data is actually coming from prometheus only from the cluster, so it being in the realfetcher makes sense since that is also for fetching any details from the cluster.